var/spack/repos/builtin/packages/ImageMagick/package.py

```diff
@@ -26,38 +26,25 @@
 
 
 class Imagemagick(Package):
-    """ImageMagick is a image processing library"""
-    homepage = "http://www.imagemagic.org"
+    """ImageMagick is a software suite to create, edit, compose,
+    or convert bitmap images."""
 
-    # -------------------------------------------------------------------------
-    # ImageMagick does not keep around anything but *-10 versions, so
-    # this URL may change.  If you want the bleeding edge, you can
-    # uncomment it and see if it works but you may need to try to
-    # fetch a newer version (-6, -7, -8, -9, etc.) or you can stick
-    # wtih the older, stable, archived -10 versions below.
-    #
-    # TODO: would be nice if spack had a way to recommend avoiding a
-    # TODO: bleeding edge version, but not comment it out.
-    # -------------------------------------------------------------------------
-    # version('6.9.0-6', 'c1bce7396c22995b8bdb56b7797b4a1b',
-    # url="http://www.imagemagick.org/download/ImageMagick-6.9.0-6.tar.bz2")
+    homepage = "http://www.imagemagick.org"
+    url      = "http://www.imagemagick.org/download/ImageMagick-7.0.2-6.tar.gz"
```

Comment:
Is there a reason we need the bleeding edge here?  Can Spack just use the
latest -10 release that is stable?  If we can't for Matplotlib... can we at
have at least one stable version in this package, and set `preferred=True`
on it?

I agree... for non-stable versions, we should fetch from GitHub (either a
`git` download, or a GitHub tar URL).  We will also need to
depends_on('autotools') and do the autoconf() thing at the beginning of
`install()` (other packages do this; search for specifics).


On Thu, Aug 11, 2016 at 2:30 PM, Adam J. Stewart <notifications@github.com>
wrote:

> In var/spack/repos/builtin/packages/ImageMagick/package.py:
>
> >
> > -    # -------------------------------------------------------------------------
> > -    # ImageMagick does not keep around anything but *-10 versions, so
> > -    # this URL may change.  If you want the bleeding edge, you can
> > -    # uncomment it and see if it works but you may need to try to
> > -    # fetch a newer version (-6, -7, -8, -9, etc.) or you can stick
> > -    # wtih the older, stable, archived -10 versions below.
> > -    #
> > -    # TODO: would be nice if spack had a way to recommend avoiding a
> > -    # TODO: bleeding edge version, but not comment it out.
> > -    # -------------------------------------------------------------------------
> > -    # version('6.9.0-6', 'c1bce7396c22995b8bdb56b7797b4a1b',
> > -    # url="http://www.imagemagick.org/download/ImageMagick-6.9.0-6.tar.bz2")
> > +    homepage = "http://www.imagemagick.org"
> > +    url      = "http://www.imagemagick.org/download/ImageMagick-7.0.2-6.tar.gz"
>
> Man, they just don't make it easy, do they. I'll try that git archive out
> and make sure it works.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1041/files/f147d645272bc96e57c707f39a80b64d1e637fb1#r74477105>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1cd-00MIdN2xT9mZZ5ulWjzr3BvHqqks5qe2pKgaJpZM4IzSyf>
> .
>